app/models/profile.rb

```diff
@@ -92,7 +92,9 @@ def activate(reason_deactivated: nil)
     confirm_that_profile_can_be_activated!
 
     now = Time.zone.now
-    is_reproof = Profile.find_by(user_id: user_id, active: true)
+    profile_to_deactivate = user.active_profile
+    is_reproof = profile_to_deactivate.present?
+    is_biometric_upgrade = is_reproof && biometric? && !profile_to_deactivate.biometric?
```

Comment:
also, this could be faster

`is_reproof = !!user.active_profile`
  is_biometric_upgrade = is_reproof && biometric? && !user.active_profile.biometric?`

Reply:
`user.active_profile` gets cached, so I am a little worried it might be easier to find a race condition in there

Reply:
wait, it shouldn't have any difference from a find right? There's no async happening...

Reply:
no just sif `user.active_profile` gets loaded and cached earlier in the process than doing the direct find later. It's probably fine?